kr8s/_portforward.py

```diff
@@ -130,21 +143,30 @@ async def run_forever(self) -> None:
         """
         async with self:
             with contextlib.suppress(asyncio.CancelledError):
-                await self.server.serve_forever()
+                for server in self.servers:
+                    await server.serve_forever()
 
     @asynccontextmanager
     async def _run(self) -> int:
-        """Start the port forward and yield the local port."""
-        self.server = await asyncio.start_server(
-            self._sync_sockets, port=self.local_port, host="0.0.0.0"
-        )
-        async with self.server:
-            await self.server.start_serving()
-            for sock in self.server.sockets:
-                if sock.family == socket.AF_INET:
-                    yield sock.getsockname()[1]
-            self.server.close()
-            await self.server.wait_closed()
+        """Start the port forward for multiple bind addresses and yield the local port."""
+        self.servers = []
+        for address in self.address:
+            server = await asyncio.start_server(
+                self._sync_sockets, port=self.local_port, host=address
```

Comment:
If `self.local_port` is `0` the server will start on a random high port. By calling this in a loop each address will be listening on a different port. We probably need to improve things here and if it is zero choose an unused port ahead of time and then reuse that same port for all addresses. This [SO question](http://stackoverflow.com/questions/2838244/get-open-tcp-port-in-python) might be useful.